### DISCUSSION TITLE
sdr++ build for Odroid N2+ running Ubuntu 22.04  - Success

### DISCUSSION THREAD

**Discussion by nigelg730** [AlexandreRouma/SDRPlusPlus#1016] [General]
Just to report the source code 22 Nov 2022 builds and installs cleanly supporting SDRPlay hardware with no errors for an Odroid N2+ running Ubuntu 22.04. Odroid N2+ is a single board computer  (4Gb RAM) running a quad core  ARM RISC CPU. 

It can be done using the published script for the Raspberry Pi build but it will want libvolk2 not the older libvolk1 for the deps.

Be patient make takes 30 minutes. 
 
However tried with source code from master  source as of 18 March 2023 and cmake ended with an error, as it failed to find the package containing libzstd  which was installed on the Odroid.  

Hop this helps anybody who wants sdr++ on a single board computer with a bit more performance than a Raspberry Pi. The onboard audio on an N2+ is very good quality with 192kbps sampling available. 

Nigel

**Comment by AlexandreRouma** [comment]
> it failed to find the package containing libzstd which was installed on the Odroid

It wasn't installed properly otherwise pkgconfig would find it.
Probably installed the non-dev version

**Reply by nigelg730** [reply]
Yes that was it. VMT.  Synaptic Package Manager confirmed state of play. 

Ubuntu 22.04 standard build for Odroid N2(+) had not included libzstd-dev package, but it did have libzstd1  
Standard Ubuntu (amd64 for me) has the libzstd-dev package already installed along with libzstd1 and libzstd1:i386. 
So if I understand it right the 22 Nov 2022 source did not need libzstd-dev since it build 'out of the box' but the master source as of 18 March 2023 does need package libzstd-dev. 

Next move:  I will install libzstd-dev package and try an Odroid N2+ Ubuntu 22.04  build on the master source. 

Thanks to Alexandre, and all who contribute, for a wonderful application, most impressed by the lack of bloat.

**Reply by HemiBob** [reply]
Nigel, I have an Odroid N2+ doing nothing, so seeing your post makes me wonder if I can use it for 24/7 NOAA reception.
Have you tried predict and wxtoimg alongside sdr++ on your Odroid?
If I know it can work, I'll have a go as I'm not doing so well on Mint on other hardware.

**Comment by nigelg730** [comment]
Hi Bob. No direct experience of NOAA decode on Odroid or predict or wxtoimg .  I have always found Ubuntu to be very very well supported on the N2(+) and I run version 22.04. In terms of max load that the N2(+) can take I did discover that running a websdr under Firefox and then wstjx  that real time decode of the more complex  processor intensive data streams like wspr was not possible. It could not complete the wspr decode before the next major frame started. Both Firefox picking up a websdr and wstjx  decoding wspr  were big loads. But sdr++ should be much lighter than Firefox. I have used fldigi  and that runs just fine on Odroid and is avaliable as a package. 

Using Ubuntu as your OS should give maximum opportunity for a working result especially as it uses the native GPU well with an excellent  driver.

**Reply by HemiBob** [reply]
Thanks Nigel, once I get my present project running, I'll have a play and see how I get on with the N2+ as it's a nice sbc just gathering dust.
